src/libraries/System.Net.Http/src/System/Net/Http/WasiHttpHandler/WasiHttpHandler.cs

```diff
@@ -14,13 +14,170 @@
 using WasiHttpWorld;
 using WasiHttpWorld.wit.imports.wasi.http.v0_2_1;
 using WasiHttpWorld.wit.imports.wasi.io.v0_2_1;
+using static WasiHttpWorld.wit.imports.wasi.http.v0_2_1.ITypes;
 
 namespace System.Net.Http
 {
+    // on top of https://github.com/WebAssembly/wasi-http/blob/main/wit/types.wit
+    internal sealed class WasiRequestWrapper : IDisposable
+    {
+        private FutureIncomingResponse? future; // owned by this instance
+        private WasiOutputStream? wasiOutputStream; // owned by this instance
+        private WasiInputStream? incomingStream; // owned by this instance
+
+        public Task? requestBodyComplete;
+        public Task<IncomingResponse>? requestComplete;
+        private bool isDisposed;
+
+        public async Task<HttpResponseMessage> SendRequestAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+        {
+            if (request.RequestUri is null)
+            {
+                throw new ArgumentException();
+            }
+
+            try
+            {
+                var requestHeaders = WasiHttpInterop.ConvertRequestHeaders(request);
+                var outgoingRequest = new OutgoingRequest(requestHeaders); // passing requestHeaders ownership
+                outgoingRequest.SetMethod(WasiHttpInterop.ConvertMethod(request.Method));
+                outgoingRequest.SetScheme(WasiHttpInterop.ConvertScheme(request.RequestUri));
+                outgoingRequest.SetAuthority(WasiHttpInterop.ConvertAuthority(request.RequestUri));
+                outgoingRequest.SetPathWithQuery(request.RequestUri.PathAndQuery);
+
+                requestBodyComplete = SendContent(request.Content, outgoingRequest, cancellationToken);
```

Comment:
AFAICT we never do anything with `requestBodyComplete`.  Is that going to be a problem?  Can we rely on it running to completion even though it's never `await`ed?  I'm used to Rust `Future`s, which are inert unless explicitly polled; I guess C# `Task`s are more like JavaScript `Promise`s in that they will make progress even if nobody's listening, but I just want to make sure I'm understanding correctly.

Also, is there any point in making it a member variable of the class rather than a local given we never do anything with it anyway?  Should we even consider ignoring the return value of `SendContent` entirely (i.e. not even use a local variable)?

Reply:
> AFAICT we never do anything with `requestBodyComplete`. 
> Also, is there any point in making it a member variable

I will clean both members, I used it for testing and forget to clean it up.

> Can we rely on it running to completion even though it's never `await`ed? 
> I guess C# `Task`s are more like JavaScript `Promise`s in that they will make progress even if nobody's listening

Yes